features/enlinkd/persistence/impl/src/main/java/org/opennms/netmgt/enlinkd/persistence/impl/TopologyEntityDaoHibernate.java

```diff
@@ -28,19 +28,45 @@
 
 package org.opennms.netmgt.enlinkd.persistence.impl;
 
+import java.sql.SQLException;
 import java.util.List;
+import java.util.stream.Collectors;
 
 import org.opennms.netmgt.enlinkd.persistence.api.TopologyEntityDao;
+import org.opennms.netmgt.model.OnmsCriteria;
+import org.opennms.netmgt.model.OnmsNode;
+import org.hibernate.Criteria;
+import org.hibernate.HibernateException;
+import org.hibernate.Session;
+import org.hibernate.criterion.CriteriaSpecification;
+import org.opennms.netmgt.dao.api.NodeDao;
 import org.opennms.netmgt.enlinkd.model.CdpLinkTopologyEntity;
 import org.opennms.netmgt.enlinkd.model.NodeTopologyEntity;
+import org.springframework.orm.hibernate3.HibernateCallback;
 import org.springframework.orm.hibernate3.support.HibernateDaoSupport;
 
 public class TopologyEntityDaoHibernate extends HibernateDaoSupport implements TopologyEntityDao {
 
     @Override
     public List<NodeTopologyEntity> getNodeTopologyEntities() {
+        /* FIXME  do not use OnmsCriteria
         return (List<NodeTopologyEntity>)getHibernateTemplate().find(
                 "select new org.opennms.netmgt.enlinkd.model.NodeTopologyEntity(n.id, n.type, n.sysObjectId, n.label, n.location) from org.opennms.netmgt.model.OnmsNode n");
+                */
+        final OnmsCriteria criteria = new OnmsCriteria(OnmsNode.class);
```

Comment:
maybe you want to test how much slower this join is compared to just reading from one table.

Reply:
consider that in this way we do not need to walk the ipinterface table....